### DISCUSSION TITLE
Visual studio - plaformio esp-idf version mismatch

### DISCUSSION THREAD

**Discussion by nliaudat** [jomjol/AI-on-the-edge-device#1569] [General]
Hi, 

I've run `pio pkg update` in vscode but as I compile I get `framework-espidf @ 3.40402.0 (4.4.2)`

```
PLATFORM: Espressif 32 (5.2.0+sha.fee4b73) > AI Thinker ESP32-CAM
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (esp-prog) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-espidf @ 3.40402.0 (4.4.2)
 - tool-cmake @ 3.16.4
 - tool-esptoolpy @ 1.40400.0 (4.4.0)
 - tool-idf @ 1.0.1
 - tool-mconf @ 1.4060000.20190628 (406.0.0)
 - tool-ninja @ 1.9.0
 - toolchain-esp32ulp @ 1.22851.191205 (2.28.51)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch3
Reading CMake configuration...
```

Anybody can help ?

I've tested latest espidf framework in plateformio.ini
`platform = https://github.com/platformio/platform-espressif32.git`

but nothing changes

I also tried [Espressif IDF extension](https://marketplace.visualstudio.com/items?itemName=espressif.esp-idf-extension)

Many thanks if you could help compile with latest espidf. 
Regards
Nicolas

**Comment by Slider0007** [comment]
@nliaudat: For me everthing looks OK, but maybe it's a little bit confusing...
You're using the latest Platformio platform-espressif32 release 5.2.0 which uses ESP-IDF v4.4.2.

Please compare: https://github.com/platformio/platform-espressif32/releases
